services/vision/obstacledepth/obstacle_depth.go

```diff
@@ -0,0 +1,376 @@
+// Package obstacledepth uses an underlying depth camera to fulfill GetObjectPointClouds,
+// using the method outlined in (Manduchi, Roberto, et al. "Obstacle detection and terrain classification
+// for autonomous off-road navigation." Autonomous robots 18 (2005): 81-102.)
+package obstacledepth
+
+import (
+	"context"
+	"image"
+	"math"
+	"runtime"
+	"strconv"
+	"sync"
+
+	"github.com/edaniels/golog"
+	"github.com/golang/geo/r3"
+	"github.com/muesli/clusters"
+	"github.com/muesli/kmeans"
+	"github.com/pkg/errors"
+	"go.opencensus.io/trace"
+
+	"go.viam.com/rdk/components/camera"
+	"go.viam.com/rdk/pointcloud"
+	"go.viam.com/rdk/resource"
+	"go.viam.com/rdk/rimage"
+	"go.viam.com/rdk/rimage/transform"
+	"go.viam.com/rdk/robot"
+	svision "go.viam.com/rdk/services/vision"
+	"go.viam.com/rdk/spatialmath"
+	"go.viam.com/rdk/utils"
+	vision "go.viam.com/rdk/vision"
+	"go.viam.com/rdk/vision/segmentation"
+)
+
+var model = resource.DefaultModelFamily.WithModel("obstacle_depth")
+
+// ObstaclesDepthConfig specifies the parameters to be used for the obstacle depth service.
+type ObstaclesDepthConfig struct {
```

Comment:
Once these are finalized, please email scope reviewers with the config parameters.